### ISSUE TITLE
Animation is playing, but mesh is not animated (random).

### ISSUE THREAD

**Issue by Yogoda** [godotengine/godot#48152]
**Godot version:**
3.3, this is working in 3.2 (regression)

**OS/device including version:**
GLES3
Windows 10
AMD Radeon R9 200 Series

**Issue description:**
I have barrels that the player can open. When opening a barrel, an animation is played that opens the lid.

In 3.2, it works correctly.
In 3.3, randomly the barrel won't open, but the animation is playing. I know the animation plays because I put a bone attachment, so I can see the bone is animating, but the mesh is not.

This is random, if I reload the map, some barrels will open, and some not, not always the same.
I have the same issue with chests.

So it seems the issue is linked to the skinning system. However, I am not using the new software skinning system in 3.3.

**Steps to reproduce:**
Unknown.

**Minimal reproduction project:**
https://1drv.ms/u/s!Ago61yJ7YVZDlE0xNRkfokNk3U4g?e=dejkFR

**Comment by Yogoda** [comment]
I was able to reproduce the issue using my zone loading system:

https://1drv.ms/u/s!Ago61yJ7YVZDlE0xNRkfokNk3U4g?e=dejkFR

There is a thread that is used to load the zones, maybe that is what causes the issue?

Working in 3.2
Not working in 3.3
Can someone confirm?

**Comment by Calinou** [comment]
Can you test this in all older 3.3 RCs and betas? You can download them [here](https://downloads.tuxfamily.org/godotengine/3.3/) and [here](https://downloads.tuxfamily.org/godotengine/3.2.4/) (3.2.4 was renamed to 3.3 during the development process).

**Comment by Yogoda** [comment]
The issue was already present in 3.2.4 beta 1.
I tested again in 3.2.3 final, and it is working correctly.

**Comment by Yogoda** [comment]
I found the exact commit that introduced this issue by compiling several versions of Godot and testing the bug.

This is the commit f9544716f432cf3c04ee71c89354d8fbca0467c2: Option for software skinning in MeshInstance
https://github.com/godotengine/godot/commit/f9544716f432cf3c04ee71c89354d8fbca0467c2

**Comment by akien-mga** [comment]
CC @pouleyKetchoupp @lawnjelly

**Comment by lawnjelly** [comment]
I can confirm seeing this both with and without software skinning (it doesn't work every time for me, it doesn't seem to be random). This will probably need @pouleyKetchoupp to look at as I'm not super familiar with the setup aspects of the PR, and indeed how it worked originally (although I got added as co-author I only helped with a limited part in the skinning code itself if I remember right).

I'll keep looking in case I spot something.

**Comment by lawnjelly** [comment]
I've managed to cut down the project to something more minimal. Just run it click 3d, and walk up to the barrel then click E to interact.

[SkinAnimationBug.zip](https://github.com/godotengine/godot/files/6371802/SkinAnimationBug.zip)

By getting rid of the other meshes and adding some debug prints what appears to happen is this:

```
1432  set_mesh
1432  attaching skeleton invalid
1432 BarrelMesh resolve_skeleton_path
1432 BarrelMesh attaching skeleton valid
```
The 1432 is the godot instance ID, after that is the name from get_name(). It seems like set_mesh is called first, without a valid skin_ref. This calls `instance_attach_skeleton` with RID(), which appears to put the instance in a state it can't get out of by calling the second time with a valid RID.

**Comment by pouleyKetchoupp** [comment]
@lawnjelly Thanks for investigating! I'm having a look right now.

**Comment by akien-mga** [comment]
Fixed by #48217.
